prevent interference with other logging by not using root logger

pawl opened the pull request:
Currently django-mailer relies on the root logger for logging, but unfortunately this can interfere with other loggers in an application. This can also unintentionally configure the root logger for users who have django configured with `'disable_existing_loggers': True` and potentially cause duplicate log messages.

This PR switches to using module specific loggers with `logger = logging.getLogger(__name__)` to resolve the issue.

This has some other advantages too, like allowing users to adjust log levels for only the django-mailer module.

spookylukey commented:
Thanks so much for looking at this. Is there a reason we are doing logging setup at all? The normal policy is to let logging get configured at project level (e.g. settings.LOGGING). I'm guessing this is a left over from a very long time ago when Django didn't have that?

pawl commented:
@spookylukey Removing the logging setup from `retry_deferred` seems safe since it only has a single `INFO` level log message. I went ahead and made a commit for that here: https://github.com/pinax/django-mailer/pull/130/commits/dd63ad9a116dda4f7b485d21fcceb4e81dbbf6d3

Removing the [line that sets ERROR level logging in](https://github.com/pinax/django-mailer/pull/130/files#diff-7e7ce70231173b80c6ad34f15f7816bb3bd5005e2b4f4e109d6b7d6925978059R23) `send_mail` should be safe too.

But I worry about removing the line in `send_mail` that sets it to `DEBUG` level, because most projects are probably not set to `DEBUG` level and that would hide some log messages that were previously showing. That `DEBUG` level logging is the default behavior when running `send_mail` without `--cron 1`.

We might need to check the current log level and show a warning like "send_mail will rely on your django project's log levels in future releases. You will need to enable logging in your django settings to continue seeing all the same logs.". Thoughts?

pawl commented:
@spookylukey I could also revert https://github.com/pinax/django-mailer/pull/130/commits/dd63ad9a116dda4f7b485d21fcceb4e81dbbf6d3 and we could defer the decision to remove the logging setup to another PR.

spookylukey commented:
I think the PR is fine, in fact I think we should remove all logger setup and just document it in the release notes, with the deprecation notice like you have done for `retry_deferred`.

pawl commented:
@spookylukey I went ahead and removed all the logger setup logic and added a note to the change log here: https://github.com/pinax/django-mailer/pull/130/commits/69cd60fdf0592889f9a21d17193d2310e88f733a

I also changed some log messages in `send_mail` (for failing to get the file lock) to error level: https://github.com/pinax/django-mailer/pull/130/commits/dc92d69c2aecb10fed8a143a08400d01a44be28d

spookylukey commented:
Thanks so much for doing this, merged now!
